src/superphot_plus/samplers/licu_sampler.py

```diff
@@ -0,0 +1,219 @@
+            ref_band=priors.reference_band,
+        )
+        sample = np.concatenate([cube, red_neg_chisq])
+
+        samples = sample.reshape(1, -1)
+
+        return PosteriorSamples(
+            samples,
+            name=lightcurve.name,
+            sampling_method="light-curve-package",
+            sn_class=lightcurve.sn_class,
+        )
+
+    def run_multi_curve(self, lightcurves, priors, **kwargs) -> List[PosteriorSamples]:
+        """Not yet implemented."""
+        raise NotImplementedError
+
+
+def fit_single_band(lc, prior_clip_a, prior_clip_b, prior_mean, **licu_kwargs):
+    """Fit a single band light curve with light_curve.VillarFit
+
+    Parameters
+    ----------
+    lc : Lightcurve
+        The light curve with a single band
+    prior_clip_a : array-like
+        The lower bound of the prior
+    prior_clip_b : array-like
+        The upper bound of the prior
+    prior_mean : array-like
+        The mean of the prior
+    licu_kwargs : dict
+        Keyword arguments to pass to the light-curve package's VillarFit
+    """
+    left_bound = transform_to_licu(*prior_clip_a)
+    # change baseline to be not exactly zero
+    left_bound[1] = -1e-3 * prior_clip_a[0]
+
+    right_bound = transform_to_licu(*prior_clip_b)
+    # change baseline to be not exactly zero
+    right_bound[1] = 1e-3 * prior_clip_a[0]
+    # relative plateau amplitude must be less than 1
+    right_bound[5] = np.min([right_bound[5], 0.99])
+
+    initial_guess = transform_to_licu(*prior_mean)
+
+    villar_fit = licu.VillarFit(  # pylint: disable=no-member
+        init=initial_guess,
+        bounds=list(zip(left_bound, right_bound)),
+        **licu_kwargs
+    )
+
+    # We have nothing better to do than just use the mean of the prior
+    extra_sigma = prior_mean[6]
+
+    # light-curve package parameters
+    # Reduced Chi^2 definition is different from the one in superphot+
+    try:
+        *features, _reduced_chi2 = villar_fit(lc.times, lc.fluxes, lc.flux_errors)
+    except ValueError:
+        features = initial_guess
+
+    cube = transform_from_licu(*features)
+    # Use mean of prior for extra_sigma
+    cube[6] = extra_sigma
+
+    return cube
+
+
+def transform_priors_to_physical_values(cube, max_flux, max_flux_loc, *, reference_idx):
+    """Some priors are for log-params, this transforms them back to linear"""
+    cube = cube.reshape(-1, 7)
+    output = cube.copy()
+
+    # Scale log-priors to linear values
+    output[:, [0, 2, 4, 5, 6]] = 10 ** output[:, [0, 2, 4, 5, 6]]
+
+    # amplitude and extra_sigma must be scaled by the maximum flux
+    output[reference_idx, [0, 6]] *= max_flux
+
+    # Non-reference band priors must be scaled by the reference band prior
+    non_ref_idx = np.delete(np.arange(output.shape[0]), reference_idx)
+    output[non_ref_idx] *= output[reference_idx]
+
+    # Reference time should not be scaled and should be set separately
+    output[:, 3] = cube[:, 3] + max_flux_loc
```

Comment:
this can probably be removed (times are shifted by max_flux_loc before sampling, so new max_flux_loc=0). If max_flux_loc is not 0, then t0 priors also have to be shifted so either both max_flux_loc shifts have to be included or neither of them.